### PR TITLE
FINERACT-1992: Multi-dlnqcy - Configuration of Installment level deli…

### PR DESCRIPTION
**Pull request by josehernandezfintecheandomx**
…nquency

## Description

Multi-dlnqcy - Configuration of Installment level delinquency, at Loan Product level we have added a flag to Enable / Disable the Installment Level Delinquency

[FINERACT-1992](https://issues.apache.org/jira/browse/FINERACT-1992)

Screenshot 2023-10-15 at 8 39 23 p m

Flag inherit from Loan Product to Loan entity at request time

Screenshot 2023-10-16 at 6 16 10 p m

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Comment by galovics** [comment]
@josehernandezfintecheandomx can we use the full name of "delinquency" instead of "dlqncy"? Other than that, looks good.

File: fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanApplicationTerms.java [hunk 1]
```diff
@@ -217,6 +217,8 @@ public final class LoanApplicationTerms {
     private boolean isScheduleExtensionForDownPaymentDisabled;
     private Money disbursedPrincipal;
 
+    private boolean enableInstallmentLevelDlnqcy;
+
```
Comment (ruchiD) [inline fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanApplicationTerms.java]: @josehernandezfintecheandomx, not sure if this configuration needs to be passed on for LoanApplicationTerms.
Also please check if Loan needs to inherit this configuration. If yes, changes needs to be implemented for the same.

Comment (josehernandezfintecheandomx) [reply fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanApplicationTerms.java]: Done! It has been updated and now It's on Loan entity level inherit from the Loan Product at request time

Screenshot 2023-10-16 at 6 16 10 p m

**Comment by josehernandezfintecheandomx** [comment]
> @josehernandezfintecheandomx can we use the full name of "delinquency" instead of "dlqncy"? Other than that, looks good.

@galovics It has been renamed as you suggested. Thanks

**Review by ruchiD — COMMENTED** [review]
hi @josehernandezfintecheandomx,
1. Please confirm whether avro schema changes are also needed for this configuration at this point.
2. Please also add required swagger changes for LoanProductApi.
3. Rest looks good to me.

**Comment by josehernandezfintecheandomx** [comment]
> hi @josehernandezfintecheandomx,
> 
> 1. Please confirm whether avro schema changes are also needed for this configuration at this point.
> 2. Please also add required swagger changes for LoanProductApi.
> 3. Rest looks good to me.

Yes all of them has been updated

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanproduct/api/LoanProductsApiResourceSwagger.java [hunk 2]
```diff
@@ -152,6 +152,8 @@ private PostLoanProductsRequest() {}
         public Boolean holdGuaranteeFunds;
         @Schema(example = "1")
         public Long delinquencyBucketId;
+        @Schema(example = "false")
+        public Boolean enableInstallmentLevelDelinquency;
         @Schema(example = "3")
```
Comment (ruchiD) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanproduct/api/LoanProductsApiResourceSwagger.java:157]: Please make similar changes for Get and Put resources also.

**Review by ruchiD — COMMENTED** [review]
LGTM

**Review by ruchiD — APPROVED** [review]
LGTM

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanproduct/api/LoanProductsApiResourceSwagger.java (hunk 2 above)
Comment (josehernandezfintecheandomx) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanproduct/api/LoanProductsApiResourceSwagger.java:157]: Done!

**Pull request merged**